### PR TITLE
ci(publish): adds publish job

### PR DESCRIPTION
**Pull request by beauraines**
Adds the publish github action. Note that the NPM_TOKEN must be set in
conjunction with this change.

### CODE REVIEW

**Comment by beauraines** [comment]
I created this PR to setup the publish github action. I'm not sure if this is overkill, because this package will so very rarely change. If this _is too much,_ I'll also submit a separate PR with just `npm run release` so that you can just manually publish.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: 'Publish to NPM'
+
+on:
+  workflow_run:
+    workflows: ['Node.js CI']
+    types: [completed]
+    branches: [master]
```
Comment (saintedlama) [inline .github/workflows/publish.yml]: Should be main in this repository

**Review by saintedlama — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (beauraines) [reply .github/workflows/publish.yml]: Fixed!

**Comment by beauraines** [comment]
@saintedlama I merged this one this morning, but it looks like the NPM_TOKEN still needs to be added to secrets

See the first job run https://github.com/saintedlama/should-release/actions/runs/4461541779/jobs/7835453138

**Pull request merged**